File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        // create a hash map
        // Our array ints are keys and their index is the value
        HashMap<Integer,Integer> table = new HashMap<>();


        for(int i = 0; i < nums.length; i++){
           int difference = target - nums[i];
           if(table.containsKey(difference)){
                return new int[] {table.get(difference), i};
           }
           else {
                table.put(nums[i], i); // put nums[i] not the difference
           }
        }

        return new int[]{};
    
    }
}

File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = 0
        r = len(nums) - 1
        return self.binarySearch(nums,l,r,target)


    def binarySearch(self, nums, l, r, target):
        if(l > r):
            return -1
        
        mid = (l + r) // 2

        if(nums[mid] == target):
            return mid
        elif(nums[mid] < target):
            return self.binarySearch(nums,mid + 1,r,target)
        else:
            return self.binarySearch(nums,l,mid - 1,target)
